src/Sarif/Writers/ResultLogJsonWriter.cs

```diff
@@ -124,38 +161,50 @@ public void WriteResult(Result result)
                 throw new ArgumentNullException("result");
             }
 
-            this.EnsureNotDisposed();
-            if (_writeState == State.Initial)
+            EnsureStateNotAlreadySet(Conditions.Disposed | Conditions.ResultsClosed);
+
+            if ((_writeConditions & Conditions.ResultsInitialized) != Conditions.ResultsInitialized)
             {
-                throw new InvalidOperationException(SarifResources.CannotWriteResultToolInfoMissing);
+                OpenResults();
             }
 
-            Debug.Assert(_writeState == State.WritingResults);
             _serializer.Serialize(_jsonWriter, result, typeof(Result));
         }
 
         public void CloseResults()
```

Comment:
So `ResultLogJsonWriter` has a public `CloseResults` method, but it's not part of the `IResultLogWriter` interface? I don't understand the reason for that design choice.